src/components/Application/ProgressBar.tsx

```diff
@@ -68,95 +79,205 @@ const ProgressBar: React.FC<ProgressBarProps> = ({
+            <List.Item
+              key={`ProgressSection_${sectionCode}_${number}`}
+              active={isActivePage(sectionCode, Number(number))}
+              onClick={() => handleChangeToPage(sectionCode, Number(number))}
+            >
+              <Grid className="page-row clickable">
+                <Grid.Column
+                  textAlign="right"
+                  verticalAlign="middle"
+                  className="progress-indicator-column"
+                >
+                  {progress ? (
+                    getIndicator(
+                      progress,
+                      checkPageIsStrict(number),
+                      isActivePage(sectionCode, Number(number)),
+                      ProgressType.page
+                    )
+                  ) : (
+                    <div className="progress-page-indicator" />
+                  )}
+                </Grid.Column>
+                <Grid.Column
+                  textAlign="left"
+                  verticalAlign="middle"
+                  className="progress-page-name-column"
+                >
+                  {pageName}
+                </Grid.Column>
+              </Grid>
+            </List.Item>
+          ))}
+        </List>
+      </div>
     )
   }
 
   const sectionsList = Object.values(sections).map(({ details, progress, pages }, index) => {
     const isStrictSection = !!strictSectionPage && strictSectionPage.sectionCode === details.code
-
-    const stepNumber = index + 1
     const { code, title } = details
     return {
-      key: `progress_${stepNumber}`,
+      key: `progress_${index}`,
       title: {
         children: (
-          <Grid>
-            <Grid.Column width={4} textAlign="right" verticalAlign="middle">
-              {progress && getIndicator(progress, isStrictSection, stepNumber)}
+          <Grid className="progress-row clickable">
+            <Grid.Column
+              textAlign="right"
+              verticalAlign="middle"
+              className="progress-indicator-column"
+            >
+              {progress && getIndicator(progress, isStrictSection, index === activeIndex)}
             </Grid.Column>
-            <Grid.Column width={12} textAlign="left" verticalAlign="middle">
-              <Header as="h4">{title}</Header>
+            <Grid.Column textAlign="left" verticalAlign="middle" className="progress-name-column">
+              {title}
             </Grid.Column>
           </Grid>
         ),
       },
       onTitleClick: () => handleChangeToPage(code, 1),
       content: {
-        content: getPageList(code, pages, isStrictSection),
+        content: getPageList(code, pages, isStrictSection, progress as ApplicationProgress),
       },
     }
   })
 
   return (
-    <Sticky as={Container} offset={135}>
-      <Header as="h4" style={inlineStyles.top}>
-        {strings.TITLE_INTRODUCTION}
-      </Header>
+    <Sticky as={Container} id="application-progress" offset={135} className="hide-on-mobile">
```

Comment:
it's not part of the changes you've made, but I would make this offset a constant at the top of this file (it can even be global constants, but at the top here would be enough for now)

Reply:
Cool, I've started a new `styleConstants` file and replaced the 135 in a couple of extra places too: 2f408884